### PR TITLE
barb alias generator will now match content between ITV and STV. Also…

### PR DESCRIPTION
**Pull request by alianos-**
…, refactoring.

### CODE REVIEW

File: src/main/java/org/atlasapi/equiv/generators/BarbAliasEquivalenceGenerator.java [hunk 1]
```diff
@@ -19,8 +17,11 @@
 import org.atlasapi.persistence.lookup.entry.LookupEntry;
 import org.atlasapi.persistence.lookup.mongo.MongoLookupEntryStore;
 
-import java.util.Objects;
-import java.util.Set;
+import com.metabroadcast.common.stream.MoreCollectors;
+import com.metabroadcast.common.stream.MoreStreams;
+
+import com.google.common.base.Optional;
```
Comment (AdamDM) [inline src/main/java/org/atlasapi/equiv/generators/BarbAliasEquivalenceGenerator.java]: is there a specific need for this or can we use java util optional?

File: src/main/java/org/atlasapi/equiv/generators/BarbAliasEquivalenceGenerator.java [hunk 2]
```diff
@@ -124,68 +129,74 @@ public BarbAliasEquivalenceGenerator(
...
+    }
+
+    /**
+     * Expand the list of aliases to include things that don't have the same namespace, but should
+     * still match. (such as originatingOwnerBcid and Bcid, or ITV and STV.
+     * We can then use the expanded list to fetch candidates that have either alias.
+     */
+    private Set<Alias> getExpandedAliases(Set<Alias> aliases) {
+        Set<Alias> expandedAliases = new HashSet<>();
+        for (Alias alias : aliases) {
```
Comment (AdamDM) [inline src/main/java/org/atlasapi/equiv/generators/BarbAliasEquivalenceGenerator.java]: consider streaming

File: src/main/java/org/atlasapi/equiv/generators/BarbAliasEquivalenceGenerator.java [hunk 3]
```diff
@@ -124,68 +129,74 @@ public BarbAliasEquivalenceGenerator(
...
-                replaceNamespace(oldAlias.getNamespace(), oldNamespacePrefix, newNamespacePrefix),
-                oldAlias.getValue()
-        );
+                getNewNamespace(alias.getNamespace(), oldPrefix, newPrefix),
+                alias.getValue());
     }
-
-    private String replaceNamespace(String namespace, String oldNamespacePrefix, String newNamespacePrefix) {
-        return newNamespacePrefix.concat(namespace.substring(oldNamespacePrefix.length()));
+    //replaces the oldPrefix with the newPrefix in the given namespace.
```
Comment (AdamDM) [inline src/main/java/org/atlasapi/equiv/generators/BarbAliasEquivalenceGenerator.java]: comment and variables all talk about prefix, maybe `getNewNamespacePrefix(..` ?

File: src/main/java/org/atlasapi/equiv/generators/BarbAliasEquivalenceGenerator.java (hunk 1 above)
Comment (alianos-) [reply src/main/java/org/atlasapi/equiv/generators/BarbAliasEquivalenceGenerator.java]: Y the lookupEntryStore requires guava. I guess it was made pre java8

File: src/main/java/org/atlasapi/equiv/generators/BarbAliasEquivalenceGenerator.java (hunk 2 above)
Comment (alianos-) [reply src/main/java/org/atlasapi/equiv/generators/BarbAliasEquivalenceGenerator.java]: meh using a foreach here would be way less readable.

File: src/main/java/org/atlasapi/equiv/generators/BarbAliasEquivalenceGenerator.java (hunk 3 above)
Comment (alianos-) [reply src/main/java/org/atlasapi/equiv/generators/BarbAliasEquivalenceGenerator.java]: It doesn't get the prefix, it replaces the oldPrefix with the newPrefix in the given namespace, and returns the namespace.

**Pull request merged**